provider/cluster_rosa_classic_resource.go

```diff
@@ -1414,7 +1465,20 @@ func populateRosaClassicClusterState(ctx context.Context, object *cmv1.Cluster,
 			Null: true,
 		}
 	}
+	channel_group, ok := object.Version().GetChannelGroup()
+	if ok {
+		state.ChannelGroup = types.String{
+			Value: channel_group,
+		}
+	} else {
+		state.ChannelGroup = types.String{
+			Null: true,
+		}
+	}
 	version, ok := object.Version().GetID()
+	// If we're using a non-default channel group, it will have been appended to
+	// the version ID. Remove it before saving state.
+	version = strings.TrimSuffix(version, fmt.Sprintf("-%s", channel_group))
```

Comment:
I found the version naming scheme of the provider to be a bit confusing... We require versions to be specified as `openshift-vX.Y.Z`, but rosa cli and ocm cli don't. Additionally, when requesting a version from a non-default channel, the channel name needs to be appended.
I chose to make that behavior transparent in this patch by auto-appending and removing as necessary instead of requiring `openshift-vX.Y.Z-channel` to be provided.

Reply:
So I think better we will be aligned with ROSA CLI and removing that suffix from the version